#!/usr/bin/env python
PACKAGE = "multi_sensor_alignment"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("i_min",         double_t, 0, "intensity lower filter limit",        0,       0,  1000.0)
gen.add("i_max",         double_t, 0, "intensity upper filter limit",    20000,       0, 20000.0)
gen.add("x_min",         double_t, 0, "x lower filter limit",             -100, -1000.0,  1000.0)
gen.add("x_max",         double_t, 0, "x upper filter limit",              100, -1000.0,  1000.0)
gen.add("y_min",         double_t, 0, "y lower filter limit",             -100, -1000.0,  1000.0)
gen.add("y_max",         double_t, 0, "y upper filter limit",              100, -1000.0,  1000.0)
gen.add("z_min",         double_t, 0, "z lower filter limit",             -100, -1000.0,  1000.0)
gen.add("z_max",         double_t, 0, "z upper filter limit",              100, -1000.0,  1000.0)
gen.add("VoxelSize",     double_t, 0, "voxel size",                       0.05,       0,       3)

gen.add("Method",                       int_t, 0, "registration method",            1,         0,        3)
gen.add("Epsilon",                   double_t, 0, "icp epsilon",                 1e-6,     1e-15,        3)
gen.add("MaxIterations",                int_t, 0, "icp maximum iterations",       100,         0,  1000000)
gen.add("MaxCorrespondenceDistance", double_t, 0, "icp max correspondence distance",  0.1,  1e-6,      100)

gen.add("KSearch",          int_t, 0, "# of neighbors to use for normal",   30,       0,    1000)
gen.add("RadiusSearch",  double_t, 0, "Normal Search Radius",              0.5,       0,    1000)

gen.add("StepSize",                  double_t, 0, "Step Size",             0.1,    1e-6,     100)
gen.add("Resolution",                double_t, 0, "Resolution",            1.0,    1e-6,     100)

exit(gen.generate(PACKAGE, "icp_align_tool_node", "icp_align_tool"))
